overlays: sc16is75x: Add generic SPI overlay

nbuchwitz opened the pull request:
Currently four nearly identical overlays (sc16is750-spi0 / -spi1 and sc16is752-spi0 / -spi1) are provided. Besides the redundant configuration all of them lack support for other SPI interfaces than spi0 and spi1.

Thus refactor the common definitions into a generic sc16is75x-spi overlay which provides support for all known spi / cs combinations. Also choose the chip type via dtparam rather than different overlay.

In order to keep existing setups keep the old overlays, but add a deprecation message via overlay map.

nbuchwitz committed:
overlays: sc16is75x: Add generic SPI overlay

Currently four nearly identical overlays (sc16is750-spi0 / -spi1
and sc16is752-spi0 / -spi1) are provided. Besides the redundant configuration
all of them lack support for other SPI interfaces than spi0 and spi1.

Thus refactor the common definitions into a generic sc16is75x-spi overlay which
provides support for all known spi / cs combinations. Also choose the chip type
via dtparam rather than different overlay.

In order to keep existing setups keep the old overlays, but add a deprecation
message via overlay map.

Signed-off-by: Nicolai Buchwitz <n.buchwitz@kunbus.com>

nbuchwitz commented:
As talked about yesterday, I will think about some generic solution in order to avoid duplication of the boilerplate spi cs tuples. But this is stuff for another PR (once I found something which will work for most of the overlays)

pelwell commented:
I'm looking at this now. The deprecation breaking the overlay check is unfortunate, but I think it would be fixed by actually deleting the original overlays. A more serious problem is the reference to labels that don't exist in the older Pis' dtbs, eg. `spi3`. This can be avoided by replacing the `target` property with `target-path=spi3` etc., making use of the fact that paths that don't start with a `/` are assumed to begin with the name of an alias, which is then resolved.

If you don't mind sticking to the old naming scheme - `int_pin` and `xtal` - we should be able to replace the `deprecated` error messages with e.g.:
```
	sc16is750-spi0 {
		rename = "sc16is75x,sc16is750,spi0-0";
	};
```
I'm currently getting an error about overlay 'sc16is750-spi0' not being supported on the bcm2835 platform, but I should be able to fix (or at least understand) that.

nbuchwitz commented:
Thanks, the changes sound resonable. I will update  the branch

pelwell commented:
I'd hang on until I've got it all working, especially the automated checks. When I've done that I'll push a patch with my changes.

pelwell committed:
SQUASH: Updates to PR 6865

As discussed, here are a few minor improvements to the original PR:
1. Use target-path with aliases to allow the overlay to be loaded on
older devices that lack the higher numbered SPI interfacs.
2. Change the "deprecated" errors to "renamed", so that the old names
continue to work without requiring a config.txt change.
3. Use the same parameter names as the old overlays, partly so that
existing installations continue to work even though they are now using
the new overlay, and partly to match the -i2c variants.
4. Actually delete the old overlays.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>

pelwell commented:
Assuming the other tests pass, we can fix up the over-long commit message when we "Squash and merge".

nbuchwitz commented:
Looks good so far. Will do some tests with real hardware and then this should be ready
